client/core/core.go

```diff
@@ -6670,3 +6672,228 @@ func (c *Core) WalletLogFilePath(assetID uint32) (string, error) {
+			cord.BaseSymbol,               // Base
+			cord.QuoteSymbol,              // Quote
+			ordReader.BaseQtyString(),     // Base Quantity
+			ordReader.SimpleRateString(),  // Order Rate
+			ordReader.AverageRateString(), // Actual Rate
+			ordReader.BaseAssetFees(),     // Base Fees
+			ordReader.QuoteAssetFees(),    // Quote Fees
+			ordReader.Type.String(),       // Type
+			ordReader.SideString(),        // Side
+			cord.TimeInForce.String(),     // Time in Force
+			ordReader.StatusString(),      // Status
+			cord.TargetOrderID.String(),   // Target Order ID
+			ordReader.FilledPercent(),     // Filled
+			ordReader.SettledPercent(),    // Settled
+			timestamp,                     // Time
+		})
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+		csvWriter.Flush()
+		err = csvWriter.Error()
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+		return nil
+	}, ordersFile.Close, nil
+}
+
+func deleteMatchFn(matchesFileStr string) (perMatchFn func(*db.MetaMatch) error, cleanUpFn func() error, err error) {
+	matchesFile, err := createFile(matchesFileStr)
+	if err != nil {
+		return nil, nil, fmt.Errorf("problem opening orders file: %v", err)
+	}
+	csvWriter := csv.NewWriter(matchesFile)
+	csvWriter.UseCRLF = runtime.GOOS == "windows"
+	err = csvWriter.Write([]string{
+		"Host",
+		"Base",
+		"Quote",
+		"Match ID",
+		"Order ID",
+		"Quantity",
+		"Rate",
+		"Swap Fee Rate",
+		"Swap Address",
+		"Status",
+		"Side",
+		"Secret Hash",
+		"Maker Swap Coin ID",
+		"Maker Redeem Coin ID",
+		"Taker Swap Coin ID",
+		"Taker Redeem Coin ID",
+		"Refund Coin ID",
+		"Time",
+	})
+	if err != nil {
+		matchesFile.Close()
+		return nil, nil, fmt.Errorf("error writing matches CSV: %v", err)
+	}
+	csvWriter.Flush()
+	err = csvWriter.Error()
+	if err != nil {
+		matchesFile.Close()
+		return nil, nil, fmt.Errorf("error writing matches CSV: %v", err)
+	}
+	return func(mtch *db.MetaMatch) error {
+		numToStr := func(n interface{}) string {
+			return fmt.Sprintf("%d", n)
+		}
+		timestamp := encode.UnixTimeMilli(int64(mtch.MetaData.Stamp)).Local().Format(time.RFC3339Nano)
+		err = csvWriter.Write([]string{
+			mtch.MetaData.DEX,                                 // Host
+			dex.BipIDSymbol(mtch.MetaData.Base),               // Base
+			dex.BipIDSymbol(mtch.MetaData.Quote),              // Quote
+			mtch.MatchID.String(),                             // Match ID
+			mtch.OrderID.String(),                             // Order ID
+			numToStr(mtch.Quantity),                           // Quantity
+			numToStr(mtch.Rate),                               // Rate
+			numToStr(mtch.FeeRateSwap),                        // Swap Fee Rate
+			mtch.Address,                                      // Swap Address
+			mtch.Status.String(),                              // Status
+			mtch.Side.String(),                                // Side
+			fmt.Sprintf("%x", mtch.MetaData.Proof.SecretHash), // Secret Hash
```

Comment:
Should probably record the secret itself too.
What do you think about the contracts as well?  Hard to know how useful they would be if this is an archived match, meaning core completed it via either refund or redeem, if the swap was broadcast.

Reply:
Sounds good, saving it all.

Reply:
`ContractTxData` is the actual raw transaction hex.  Don't need that saved.